lib/batali/score_keeper.rb

```diff
@@ -23,51 +27,59 @@ def score_for(unit, idx)
         m_unit.name == unit.name
       end
-      if(manifest_unit)
-        # If the unit version matches the manifest version, this
-        # should be _the_ preferred version
-        if(manifest_unit.version == unit.version)
-          multiplier = 10000000
-        else
-          # If the unit version satisfies within the patch segment of
-          # the manifest version score those versions highest for upgrade
-          if(UnitRequirement.new("~> #{manifest_unit.version}").satisfied_by?(unit.version))
-            multiplier = 1000000
-          else
-            # If the unit version satisfies within the minor or major
-            # version segments of the manifest version, bump score
-            # value up (with satisfaction within minor segment being
-            # worth more than satisfaction within major segment)
-            pos = UnitRequirement.new("~> #{manifest_unit.version.segments.slice(0,2).join('.')}").satisfied_by?(unit.version) ? 1 : 0
-            multi_val = pos == 1 ? 1000 : 100
-            distance = (manifest_unit.version.segments[pos] - unit.version.segments[pos])
-            if(distance > 0)
-              distance = 1.0 / distance
-            else
-              distance = 0
-            end
-            multiplier = multi_val + (multi_val * distance)
-          end
-        end
-      end
-      score = []
-      # Generate a "value" for each segment of the version with
-      # growing importance (major > minor > patch)
-      unit.version.segments.reverse.each_with_index.map do |val, pos|
-        if(val == 0)
-          score.push 0
-        else
-          score.push (2 - (1.0 / val)) * ((pos + 1)**10)
-        end
-      end
+
+      multiplier = score_multiplier(unit, manifest_unit)
+      score_arr = version_segment_score(unit, multiplier)
       # Sum the score for each segment to provide the score for the
       # version and multiply by defined multiplier to force higher
       # score when manifest drives desire for versions closer to
       # version defined within manifest
-      score = score.inject(&:+) * multiplier
+      score = score_arr.inject(&:+) * multiplier
       debug "Score <#{unit.name}:#{unit.version}>: #{score}"
       score
     end
 
+    private
+
+    # Create a score based on the version segments (major, minor, patch)
+    #   Manifest will create a higher score for versions closest to specification
+    #
+    # @param unit [Unit] unit to score
+    # @param multiplier [Numeric] weight based on version constraint
+    # @return [Numeric]
+    def score_multiplier(unit, manifest_unit)
+      return SCORE_WEIGHT_MIN if manifest_unit.nil?
+      # If the unit version matches the manifest version, this
+      # should be _the_ preferred version
+      return SCORE_WEIGHT_MAX if manifest_unit.version == unit.version
+      return SCORE_WEIGHT_MID if UnitRequirement.new("~> #{manifest_unit.version}").satisfied_by?(unit.version)
```

Comment:
These are kind of like guards and I think it's really easy to read, since you can just go down through the cases... seemed clean?  

You feel strongly about that change? Cause I think this is pretty concise / comprehensible as is